src/_guides/language/effective-dart/design_migrated.md

```diff
@@ -785,6 +785,11 @@ Of course, it is often useful to have mutable data. But, if you don't need it,
 your default should be to make fields and top-level variables `final` when you
 can.
 
+Sometimes an instance field doesn't change after it has been initialized, but
+can't be initialized until after the instance is constructed. For example, it
+may need to reference `this` or some other field on the instance. In cases like,
+that, consider making the field `late final`.
```

Comment:
```suggestion
that, consider making the field `late final`. When a field is `late final` it may also be possible to use an initializing expression on the field declaration.
```

Reply:
Good suggestion. I'll reword this a little and apply it to my change.